api/method_assertions.go

```diff
@@ -2,14 +2,32 @@ package api
 
 import "net/http"
 
-func listAssertionsHandler(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusNotImplemented)
-	if _, err := w.Write([]byte("not implemented")); err != nil {
-		log.Error("failed to write response body", "err", err)
+func (s *Server) listAssertionsHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	ah, err := s.assertions.LatestCreatedAssertionHashes(ctx)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+
+	resp := make([]*Assertion, len(ah))
+
+	for idx, h := range ah {
+		aci, err := s.assertions.ReadAssertionCreationInfo(ctx, h)
```

Comment:
Each of these retrieves an individual event log from Ethereum. In the future, might be better to have a different strategy for performance / reducing pressure on the node. Approving for now

Reply:
Yes, we should batch these requests in a follow up PR